spack-repo/packages/singularity-eos_deps/package.py

```diff
@@ -10,29 +10,113 @@ class SingularityEosDeps(BundlePackage):
 
     version("main", branch="main")
 
-    variant("use_cuda", default=False, description="Enable cuda support")
+    variant(
+            "cuda",
+            default=False,
+            description="Use cuda"
+    )
+
+    variant(
+            "kokkos",
+            default=False,
+            description="Use kokkos"
+    )
+
+    variant(
+            "linalg", 
+            default="eigen", 
+            description="Linear algebra library",
+            values=("eigen" ,"kokkos"),
+            multi=False
+    )
+    variant(
+            "build_extra",
+            values=("sesame","stellarcollapse", "none"), 	
+            default="none",
+            description="Build converters",
+            multi=True
+    )
+
+    variant(
+            "enable_tests",
+            default=False,
+            description="Build tests"
+    )
+
+    depends_on("hdf5~mpi+cxx+hl")
+    depends_on("cuda@11:", when="+cuda")
+    depends_on("kokkos@3:", when="+kokkos")
+    depends_on("eigen@3.3.9", when="linalg=eigen")
+    depends_on("kokkos-kernels", when="linalg=kokkos")
+    depends_on("mpark-variant")
+    depends_on("eospac")
+
+    depends_on("cmake@3.12:")
+    depends_on("catch2@2.13.4:2.13.6")
+    depends_on(
+            "mpark-variant",
+#            patches=patch("cuda_compatibility.patch"),
+            when="+cuda"
+    )
+
+    conflicts(
+            "linalg=eigen",
+            when="+cuda"
+    )
+    
+#    patch("cuda_compatibility.patch", when="+cuda")
```

Comment:
And we might as well always apply the patch.

Reply:
Spack didn't it when I was using the patch, I'll see if I can get it working